### PR TITLE
problem definition

### PR DESCRIPTION
**Pull request by b5**
**DO NOT MERGE**

As mentioned in #1 & #3, there's a break in this package, this PR hopefully helps making tracking down the issue easier, this PR makes some sensible guesses, and lifts the test suite up for easier checking.

### CODE REVIEW

**Comment by whyrusleeping** [comment]
@b5 thanks for this, I do think that the lack of using Reset is a big part of the problem. We fixed a bug in yamux this way recently. cc @stebalien

File: spdystream.go [hunk 1]
```diff
@@ -36,6 +36,10 @@ func (s *stream) Close() error {
 	return s.spdyStream().Reset()
```
Comment (Stebalien) [inline spdystream.go]: This should call `Close` now.

**Review by Stebalien — COMMENTED** [review]
We'll need to fix this eventually (i.e., a month ago) but I shudder to think of the fallout...

**Comment by Stebalien** [comment]
I believe package.json needs an update hash for go-stream-muxer. Have you tested with `gx test`?

**Comment by b5** [comment]
ran `gx test`, no errors reported, but the package refs have definitely changed. If I had the time I'd dig through the gx docs, but quick instructions on how  to switch `github.com/jbenet/go-stream-muxer` -> `github.com/libp2p/go-stream-muxer`, I can ship quicker. I'm guessing `gx publish` only happens after merge.

**Comment by Stebalien** [comment]
`gx update github.com/libp2p/go-stream-muxer` should be sufficient.

**Comment by b5** [comment]
If I'm not mistaken, this might actually be mergeable? If not I can close it out. Thanks!

**Comment by Stebalien** [comment]
@whyrusleeping is on vacation at the moment and is the only one who can perform merges on this repo.

**Comment by b5** [comment]
So glad to hear Jeremy's getting some well earned time off. No rush at all!

**Comment by whyrusleeping** [comment]
I'm back! Let's go go go!

**Pull request merged**